### ISSUE TITLE
Un build plus simple ?

### ISSUE THREAD

**Issue by hgwood** [Zenika/Formation--Framework#117]
#77 et #111 me font penser que nous souffrons du choix de grunt et gulp pour le build des formations. grunt a été complètement délaissé par le communauté et nous avons de plus en plus de warning à cause des dépendances obsolètes. gulp a du mal à faire sa transition vers la version 4. Dans les 2 cas il faut maintenir le fichier de config. Il me semble que nos besoins pourrait être couvert par des scripts npm et/ou JS plus robustes.

Qu'en pensez-vous ?

Les besoins que j'ai vu (aidez à compléter) :
- serveur HTTP pour servir les slides (browser-sync ?)
- copier des fichiers (cp ? ncp ?)
- remplacer des variables dans des fichiers (template strings ES6 ?)
- générer les PDF (scripts JS ?)

**Comment by fabienjuif** [comment]
J'aurais tendance à donner le 👍 mais après en avoir parlé à @gmembre-zenika il s'avère que ce n'est pas si simple.

## Les utilisateurs de node/npm
Je suis carrement pour se passer de gulp/grunt.
Voir faire des scripts npm qui lancent docker, je le fais très souvent.
Mais il y a ... 

## Les autres
Comment lancent-il le projet ? 

### Via docker directement
Il n'y a alors pas de raccourcit, il faut taper la ligne de commande complète.
Donc ce sera peu utilisé.

### Via un script `run.sh` qui lance docker
Et là il y a un soucis : ce n'est pas maintenu dans le framework, chaque formation doit avoir son propre `run.sh`

Je trouve qu'il y a un petit soucis dans le modèle.
Je ne sais pas comment le résoudre sans "forcer" l'installation de node/npm.
Après ça peut être une issue à part, je ne sais pas si c'est lié, ça peut l'être si vous voyez une solution transverse.

**Comment by hgwood** [comment]
Les problèmes que tu soulèves sont valides mais orthogonaux à cette issue.

Le périmètre de cette issue est le système de build fourni par le framework. Il n'y pas d'impact sur la façon de lancer les formations avec Docker et `run.sh`. Seul le `Dockerfile` serait impacté.

L'amélioration "pas de `run.sh` par formation" peut faire l'objet d'une autre issue.

**Comment by jlandure** [comment]
Yep ! même sentiment que @hgwood... Je considère dans cette issue qu'on a node en pré-requis.
Pour la tambouille `docker`+`run.sh` ou autre, c'est une autre discussion.
